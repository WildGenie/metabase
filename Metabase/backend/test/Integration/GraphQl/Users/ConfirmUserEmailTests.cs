using FluentAssertions;
using System.IO;
using System.Threading.Tasks;
using Snapshooter.Xunit;
using Xunit;
using System.Collections.Generic;

namespace Metabase.Tests.Integration.GraphQl.Users
{
    public sealed class ConfirmUserEmailTests
      : UserIntegrationTests
    {
        [Fact]
        public async Task ValidData_ConfirmsUserEmail()
        {
            // Arrange
            var confirmationCode = await RegisterUser();
            // Act
            var response = await SuccessfullyQueryGraphQlContentAsString(
                File.ReadAllText("Integration/GraphQl/Users/ConfirmUserEmailTests/ValidData_ConfirmsUserEmail.graphql"),
                variables: new Dictionary<string, object?>
                {
                    ["confirmationCode"] = confirmationCode
                }
                ).ConfigureAwait(false);
            // Assert
            Snapshot.Match(
                response,
                matchOptions => matchOptions.Assert(fieldOptions =>
                 fieldOptions.Field<string>("data.confirmUserEmail.user.id").Should().NotBeNullOrWhiteSpace()
                 )
                );
        }

        [Fact]
        public async Task UnknownUser_IsUserError()
        {
            // Arrange
            var confirmationCode = await RegisterUser();
            // Act
            var response = await SuccessfullyQueryGraphQlContentAsString(
                File.ReadAllText("Integration/GraphQl/Users/ConfirmUserEmailTests/UnknownUser_IsUserError.graphql"),
                variables: new Dictionary<string, object?>
                {
                    ["confirmationCode"] = confirmationCode
                }
                ).ConfigureAwait(false);
            // Assert
            Snapshot.Match(response);
        }

        [Fact]
        public async Task InvalidConfirmationCode_IsUserError()
        {
            // Arrange
            var confirmationCode = await RegisterUser();
            // Act
            var response = await SuccessfullyQueryGraphQlContentAsString(
                File.ReadAllText("Integration/GraphQl/Users/ConfirmUserEmailTests/InvalidConfirmationCode_IsUserError.graphql"),
                variables: new Dictionary<string, object?>
                {
                    ["confirmationCode"] = "invalid" + confirmationCode
                }
                ).ConfigureAwait(false);
            // Assert
            Snapshot.Match(response);
        }
    }
}
