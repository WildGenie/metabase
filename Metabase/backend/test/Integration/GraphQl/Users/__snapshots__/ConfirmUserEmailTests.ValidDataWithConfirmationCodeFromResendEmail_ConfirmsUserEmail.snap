{"data":{"confirmUserEmail":{"user":{"id":"VXNlcgpnODM1MzQzMTdmNzJiNDk2NGI1YjFjMTU0ODFkMDIzM2U=","email":"john.doe@ise.fraunhofer.de"},"errors":null}}}
